### PR TITLE
Fix calendar book

### PR DESCRIPTION
**Pull request by jazzie-b**
Book and Calendar files have been updated to match with style guide.
Any file associated with these changes has also been updated to ensure no errors.

### CODE REVIEW

File: src/Book.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        this.title = title;
+        this.callNo = callNo;
+        this.id = id;
+        this.state = State.AVAILABLE;
+    }
+
+    
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("Book: ").append(id).append("\n")
```
Comment (pyroib) [inline src/Book.java]: change variable 'sb' to be descriptive; IE; stringBuilder

File: src/Book.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return id;
+    }
+
+    
+    public String getTitle() {
+        return title;
+    }
+
+    
+    public boolean available() {
```
Comment (pyroib) [inline src/Book.java]: consider a more descriptive method name

File: src/Book.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        return state == State.AVAILABLE;
+    }
+
+    
+    public boolean onLoan() {
+        return state == State.ON_LOAN;
+    }
+
+    
+    public boolean damaged() {
```
Comment (pyroib) [inline src/Book.java]: consider a more descriptive method name

File: src/Book.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        return state == State.DAMAGED;
+    }
+
+    
+    public void borrow() {
```
Comment (pyroib) [inline src/Book.java]: consider a more descriptive method name

File: src/Book.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        if (state.equals(State.AVAILABLE)) {
+          state = State.ON_LOAN;
+        } else {
+          throw new RuntimeException(String.format("Book: cannot borrow while book is in state: %s", state));
+        }
+    }
+
+    
+    public void returned(boolean DAMAGED) {
```
Comment (pyroib) [inline src/Book.java]: consider a more descriptive method name

File: src/Book.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          } else {
+            state = State.AVAILABLE;
+          }
+        } else {
+          throw new RuntimeException(String.format("Book: cannot Return while book is in state: %s", state));
+        }
+    }
+
+    
+    public void repair() {
```
Comment (pyroib) [inline src/Book.java]: consider a more descriptive method name

File: src/Book.java (hunk 1 above)
Comment (jazzie-b) [reply src/Book.java]: Will be changed next commit

File: src/Book.java (hunk 2 above)
Comment (jazzie-b) [reply src/Book.java]: Unchanged, will be called as book.available. I consider book.bookAvailable redundant use of word book. Same for all similar issues below.

File: src/Book.java (hunk 6 above)
Comment (jazzie-b) [reply src/Book.java]: Not changed, method called with object, eg bookObject.repair, is valid.

File: src/Book.java (hunk 5 above)
Comment (jazzie-b) [reply src/Book.java]: Not changed, method called with object, eg bookObject.returned, is valid

File: src/Book.java (hunk 4 above)
Comment (jazzie-b) [reply src/Book.java]: Not changed, method called with object, eg bookObject.borrow, is valid

File: src/Book.java (hunk 3 above)
Comment (jazzie-b) [reply src/Book.java]: Changed to isDamaged

File: src/Book.java (hunk 2 above)
Comment (jazzie-b) [reply src/Book.java]: Changed to isAvailable

**Comment by jazzie-b** [comment]
Files adjusted so merge can be handled successfully in the case Meraj's code doesn't be fixed in time

File: src/Book.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        this.callNo = callNo;
+        this.id = id;
+        this.state = State.AVAILABLE;
+    }
+
+    
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Book: ").append(id).append("\n")
+        .append("  Title:  ").append(title).append("\n")
```
Comment (pyroib) [inline src/Book.java]: check indentation

File: src/Book.java (hunk 4 above)
Comment (pyroib) [reply src/Book.java]: relying on external objects to accurately identify the scope of a method is not a good idea.

File: src/Book.java (hunk 5 above)
Comment (pyroib) [reply src/Book.java]: relying on external objects to accurately identify the scope of a method is not a good idea.

File: src/Book.java (hunk 6 above)
Comment (pyroib) [reply src/Book.java]: relying on external objects to accurately identify the scope of a method is not a good idea.

File: src/Book.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    
+    public void returned(boolean DAMAGED) {
+        if (state.equals(State.ON_LOAN)) {
+          if (DAMAGED) {
+            state = State.DAMAGED;
+          } else {
+            state = State.AVAILABLE;
+          }
+        } else {
+          throw new RuntimeException(String.format("Book: cannot Return while book is in state: %s", state));
```
Comment (pyroib) [inline src/Book.java]: remove method call within the arguments list of a method call

File: src/Book.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    public boolean isDamaged() {
+        return state == State.DAMAGED;
+    }
+
+    
+    public void borrow() {
+        if (state.equals(State.AVAILABLE)) {
+          state = State.ON_LOAN;
+        } else {
+          throw new RuntimeException(String.format("Book: cannot borrow while book is in state: %s", state));
```
Comment (pyroib) [inline src/Book.java]: remove method call within the arguments list of a method call

File: src/Book.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+          throw new RuntimeException(String.format("Book: cannot Return while book is in state: %s", state));
+        }
+    }
+
+    
+    public void repair() {
+        if (state.equals(State.DAMAGED)) {
+          state = State.AVAILABLE;
+        } else {
+          throw new RuntimeException(String.format("Book: cannot repair while book is in state: %s", state));
```
Comment (pyroib) [inline src/Book.java]: remove method call within the arguments list of a method call

File: src/Book.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+@SuppressWarnings("serial")
+public class Book implements Serializable {
+
+    private String title;
+    private String author;
+    private String callNo;
+    private int id;
+    
+    private enum State {AVAILABLE, ON_LOAN, DAMAGED, RESERVED};
+    private State state;
```
Comment (pyroib) [inline src/Book.java]: change State to be more descriptive

File: src/Book.java (hunk 4 above)
Comment (jazzie-b) [reply src/Book.java]: I will look into it again tonight and take on board those suggestions

File: src/Book.java (hunk 7 above)
Comment (jazzie-b) [reply src/Book.java]: On the stringbuilder?

File: src/Book.java (hunk 9 above)
Comment (jazzie-b) [reply src/Book.java]: For the string formatter?

File: src/Book.java (hunk 11 above)
Comment (jazzie-b) [reply src/Book.java]: Thank you for noticing that. Will adjust as requested

File: src/Book.java (hunk 9 above)
Comment (pyroib) [reply src/Book.java]: yes please.

File: src/Book.java (hunk 7 above)
Comment (pyroib) [reply src/Book.java]: yes please. 


See 4.5.2 Indent continuation lines at least +4 spaces

When line-wrapping, each line after the first (each continuation line) is indented at least +4 from the original line.

When there are multiple continuation lines, indentation may be varied beyond +4 as desired. In general, two continuation lines use the same indentation level if and only if they begin with syntactically parallel elements.

**Comment by jazzie-b** [comment]
All fixes should be applied.

**Comment by jazzie-b** [comment]
Hopefully now all fixed.

File: src/Book.java [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    public boolean isDamaged() {
+        return bookState == BookState.DAMAGED;
+    }
+
+    
+    public void borrowBook() {
+        if (bookState.equals(BookState.AVAILABLE)) {
+          bookState = BookState.ON_LOAN;
+        } else {
+          throw new RuntimeException(String.format("Book: cannot borrow while book is in state: %s", bookState));
```
Comment (pyroib) [inline src/Book.java]: check this one. repairBook is perfect :)

File: src/Book.java [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+    
+    public void returnBook(boolean DAMAGED) {
+        if (bookState.equals(BookState.ON_LOAN)) {
+          if (DAMAGED) {
+            bookState = BookState.DAMAGED;
+          } else {
+            bookState = BookState.AVAILABLE;
+          }
+        } else {
+          throw new RuntimeException(String.format("Book: cannot Return while book is in state: %s", bookState));
```
Comment (pyroib) [inline src/Book.java]: check this one. repairBook is perfect :)

**Comment by jazzie-b** [comment]
Other two exceptions should be adjusted now

**Comment by jazzie-b** [comment]
Removal via bash didn't work. library and member properly told good bye.

**Pull request merged**